hooks/useInventory.ts

```diff
@@ -0,0 +1,51 @@
+import { useState, useEffect, useCallback } from 'react';
+import { InventoryItem, ItemAndCount } from '@/database/models';
+import { readInventory, updateInventory } from '@/database/queries';
+
+const useInventory = () => {
+    const [inventory, setInventory] = useState<ItemAndCount[]>([]);
+    const [refresh, setRefresh] = useState<number>(0);
+
+    const fetchInventory = useCallback(async () => {
+        const fetchedInventory = await readInventory();
+        setInventory(fetchedInventory);
+    }, []);
+
+    useEffect(() => {
+        fetchInventory();
+    }, [fetchInventory, refresh]);
+
+    const refreshInventory = () => {
+        setRefresh(prev => prev + 1);
+    };
```

Comment:
Sorry, but this is not a good way to approach this. Probably a context is what's needed here. We want the inventory to update as needed, not need to call a refresh function.